### PR TITLE
@orta => Network requests online while online

### LINKED ISSUE #258: Starting app offline doesn't retrieve artworks
## Steps to reproduce

1. Make sure device is offline
2. Start app
3. Connect device to Internet

## Expected results

The app detects that it is now connected and downloads initial results

## Actual Results

The spinner appears but isn't animated. Nothing else happens.

### PR DESCRIPTION
**Pull request by ashfurrow**
This fixes #258. But I figured, why stop at the listings controller? 

So I used signal chaining to require network operations to be online first, in the same way we make sure we have a valid XApp token.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ before_install:
   - echo "machine github.com login $GITHUB_API_KEY" > ~/.netrc
   - chmod 600 ~/.netrc
   - xcrun instruments -w "iPad Air (8.1 Simulator)" || echo "(Pre)Launched the simulator."
+  - gem update bundler
```
Comment (orta) [inline .travis.yml]: this feels weird

**Comment by orta** [comment]
;)

not seeing any tests

File: .travis.yml (hunk 1 above)
Comment (ashfurrow) [inline .travis.yml]: Yeah I'm just trying something. It's not working :\

**Comment by ashfurrow** [comment]
I'll add some.

**Comment by ashfurrow** [comment]
Test'd!

**Comment by ashfurrow** [comment]
@orta :green_apple:

**Comment by orta** [comment]
rockkkin

**Pull request merged**